jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java

```diff
@@ -114,6 +118,38 @@ public static boolean isCRIUSupportEnabled() {
 		return criuSupportEnabled;
 	}
 
+	private String checkPointDir;
+	private boolean keepRunning = false;
+	private boolean shellJob = true;
+	private boolean extUnixSupport = true;
+	private int logLevel = 4;
+	private String logFile = "checkpoint.log";
+
+	public CRIUSupport setCheckPointDir(String checkPointDir) {
+		this.checkPointDir = checkPointDir;
+		return this;
+	}
+	public CRIUSupport setKeepRunning(boolean keepRunning) {
+		this.keepRunning = keepRunning;
+		return this;
+	}
+	public CRIUSupport setShellJob(boolean shellJob) {
+		this.shellJob = shellJob;
+		return this;
+	}
+	public CRIUSupport setExtUnixSupport(boolean extUnixSupport) {
+		this.extUnixSupport = extUnixSupport;
+		return this;
+	}
+	public CRIUSupport setLogLevel(int logLevel) {
+		this.logLevel = logLevel;
+		return this;
+	}
+	public CRIUSupport setLogFile(String logFile) {
```

Comment:
will also need the security check here if path is not null

Reply:
so no security check is needed, as we would have already done that when setting the imageDir

Reply:
I don't think that's a fair assumption, `logFile` might be an absolute or relative path that refers to some location that is not writable.

Reply:
> The logfile is stored in the workDir (defaults to imageDir) but can be set with criu_local_set_work_dir_fd. So we will need to add that.

So we will need the security check when setting workDir and imageDir, but log file will just be a file name.